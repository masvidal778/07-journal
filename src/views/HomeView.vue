<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
  </div>
  <h1>Hola, m√≥n!</h1>
  <button class="btn btn-primary">Primary</button>
  <button class="btn btn-secondary">Secondary</button>
  <button  class="btn btn-success">Success</button>
</template>

<script>
// @ is an alias to /src

export default {
  name: 'HomeView',
  components: {

  }
}
</script>

<script setup>
</script>