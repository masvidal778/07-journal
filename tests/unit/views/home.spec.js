import HomeView from "@/views/HomeView.vue";
import { shallowMount } from "@vue/test-utils";

describe("proves al Home View", () => {

    test("Renderitza correctament el component", () => {

        const wrapper  = shallowMount( HomeView )
        expect( wrapper.html() ).toMatchSnapshot()

    })

    test("Fer click en un botó ha de redirigir a no-entry", () => {

        const mockRouter = {
            push: jest.fn() //funció de jest, per saber si ha estat cridat
        }

        const wrapper = shallowMount( HomeView, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry'})

    })
})