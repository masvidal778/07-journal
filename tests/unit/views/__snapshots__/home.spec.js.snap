// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proves al Home View Renderitza correctament el component 1`] = `
<div class="home"><img alt="Vue logo" src=""></div>
<h1>Hola, m√≥n!</h1>
<button class="btn btn-primary">Primary</button>
<button class="btn btn-secondary">Secondary</button>
<button class="btn btn-success">Success</button>
`;
